js/memory.js

```diff
@@ -1,9 +1,39 @@
 class MemoryGame {
-  constructor(cards){
-    this.cards = cards;
-    // add the rest of the class properties here
+  constructor(cards) {
+    this.cards = cards
+    this.pickedCards = []
+    this.pairsClicked = 0
+    this.pairsGuessed = 0
   }
-  shuffleCards() {}
-  checkIfPair(card1, card2) {}
-  isFinished() {}
+
+
+  shuffleCards() {
+    let currentIndex = this.cards.length, temporaryValue, randomIndex
+    while (0 !== currentIndex) {
+      randomIndex = Math.floor(Math.random() * currentIndex)
+      currentIndex -= 1
+      temporaryValue = this.cards[currentIndex]
+      this.cards[currentIndex] = this.cards[randomIndex]
+      this.cards[randomIndex] = temporaryValue
+    }
+  }
+
+
+  checkIfPair(card1, card2) {
+    this.pairsClicked++
+    if (card1 === card2) {
+      return this.pairsGuessed++, true
+    } else {
+      return false
+    }
+  }
+
+  isFinished() {
+    if (this.pairsGuessed <= 12) {
+      return false
+    } else {
+      return true
+    }
+  }
```

Comment:
```suggestion
  isFinished() {
    if (this.pairsGuessed <this.cards.length / 2) {
      return false
    } else {
      return true
    }
  }
```

Reply:
Si bien con 12 te funcionaría para el lab (que usa 24 cartas), Jasmine hace los test con un juego de 16 cartas. Por eso te lo está poniendo en rojo. 
Al trabajar con this.cards.length nos aseguramos de que funcione para cualquier numero de cartas